components/concept/storage/src/main/java/mozilla/components/concept/storage/StorageMaintenanceWorker.kt

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.storage
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+/**
+ * An abstract WorkManager Worker class that executes maintenance task provided via [operate].
+ *
+ * If there is a failure or the constraints of worker are no longer met during execution,
+ * cancellation tasks are executed via [onError].
+ */
+abstract class StorageMaintenanceWorker(context: Context, params: WorkerParameters) :
+    CoroutineWorker(context, params) {
+
+    @Suppress("TooGenericExceptionCaught")
+    override suspend fun doWork(): Result = withContext(Dispatchers.IO) {
+        try {
+            operate()
```

Comment:
Didn't catch this earlier, sorry - since we have this `operate` here that we want to be overridden. It might be good to make this `doWork` method `final`?